ext/phar/config.m4

```diff
@@ -5,7 +5,17 @@ PHP_ARG_ENABLE([phar],
   [yes])
 
 if test "$PHP_PHAR" != "no"; then
-  PHP_NEW_EXTENSION(phar, util.c tar.c zip.c stream.c func_interceptors.c dirstream.c phar.c phar_object.c phar_path_check.c, $ext_shared,, -DZEND_ENABLE_STATIC_TSRMLS_CACHE=1)
+  dnl Add -Wno-empty-body because the code generated by rec2 produces this warning
```

Comment:
it would be better to change the definition of the YYFILL macro. A conservative choice that keeps current behavior is `do {} while (0)`. It might be better to convert it into an assertion, but I'm not totally sure on that.

Reply:
So defining YYFILL as ``#define YYFILL(n)       do {} while (0)`` changes the compile warning to:
```
ext/phar/phar_path_check.c: In function ‘phar_path_check’:
ext/phar/phar_path_check.c:48:41: warning: suggest braces around empty body in an ‘if’ statement [-Wempty-body]
  if ((YYLIMIT - YYCURSOR) < 4) YYFILL(4);
                                         ^
```
(note the column of the error went from 24 to 41)

So that's sort of an improvement I guess but doesn't fix the issue yet.

Any other idea?